include/core/shape.h

```diff
@@ -936,12 +2182,185 @@ class CompositeShape : public Shape<dim>
       {
         this->effective_radius =
           std::max(this->effective_radius, constituent->effective_radius);
         constituent->set_part_of_a_composite(true);
+        // For each of the components update the bounding box.
+        if constexpr (dim == 2)
+          {
+            std::vector<Point<3>> component_bounding_box_vertex(4);
+
+            component_bounding_box_vertex[0][0] =
+              constituent->get_bounding_box_center()[0] +
+              constituent->get_bounding_box_half_length()[0];
+            component_bounding_box_vertex[0][1] =
+              constituent->get_bounding_box_center()[1] +
+              constituent->get_bounding_box_half_length()[1];
+            component_bounding_box_vertex[1][0] =
+              constituent->get_bounding_box_center()[0] -
+              constituent->get_bounding_box_half_length()[0];
+            component_bounding_box_vertex[1][1] =
+              constituent->get_bounding_box_center()[1] +
+              constituent->get_bounding_box_half_length()[1];
+            component_bounding_box_vertex[2][0] =
+              constituent->get_bounding_box_center()[0] -
+              constituent->get_bounding_box_half_length()[0];
+            component_bounding_box_vertex[2][1] =
+              constituent->get_bounding_box_center()[1] -
+              constituent->get_bounding_box_half_length()[1];
+            component_bounding_box_vertex[3][0] =
+              constituent->get_bounding_box_center()[0] +
+              constituent->get_bounding_box_half_length()[0];
+            component_bounding_box_vertex[3][1] =
+              constituent->get_bounding_box_center()[1] -
+              constituent->get_bounding_box_half_length()[1];
+            for (unsigned int i = 0; i < 4; ++i)
+              {
+                // Position the vertex point in the reference frame of the
+                // composite
+                auto new_point = constituent->get_rotation_matrix() *
+                                   component_bounding_box_vertex[i] +
+                                 point_nd_to_3d(constituent->get_position());
+                if (xyz_min_max_is_initialized == false)
+                  {
+                    x_max                      = new_point[0];
+                    x_min                      = new_point[0];
+                    y_max                      = new_point[1];
+                    y_min                      = new_point[1];
+                    xyz_min_max_is_initialized = true;
+                  }
+                else
+                  {
+                    x_max = std::max(x_max, new_point[0]);
+                    x_min = std::min(x_min, new_point[0]);
+                    y_max = std::max(y_max, new_point[1]);
+                    y_min = std::min(y_min, new_point[1]);
+                  }
+              }
+          }
+        else
+          {
+            std::vector<Point<3>> component_bounding_box_vertex(8);
```

Comment:
I don't undertand why this part is not done in a loop. It just seems like copy pasted code